docker/src/main/java/org/jclouds/docker/compute/strategy/DockerComputeServiceAdapter.java

```diff
@@ -237,15 +240,21 @@ public DockerComputeServiceAdapter(DockerApi api) {
    }
 
    @Override
-   public Image getImage(final String imageId) {
+   public Image getImage(final String imageIdOrName) {
       // less efficient than just inspectImage but listImages return repoTags
-      return find(listImages(), new Predicate<Image>() {
-
+      Optional<Image> image = tryFind(listImages(), new Predicate<Image>() {
          @Override
          public boolean apply(Image input) {
-            return input.id().equals(imageId);
+            // Only attempt match on id as we should try to pull again anyway if using name
+            return input.id().equals(imageIdOrName);
          }
-      }, null);
+      });
+      if (image.isPresent()) return image.get();
+
+      // Image is not cached or getting image by name so try to pull it
+      //
+      api.getImageApi().createImage(CreateImageOptions.Builder.fromImage(imageIdOrName));
```

Comment:
Sorry - for some reason I'd confused myself and decided that would work when the api docs make it clear that won't - I'll change and remove the unused imports that I seem to have left in.